intents.go

```diff
@@ -27,111 +44,122 @@ const (
 	Expired      ProcessingStatus = "Expired"
 	OnChain      ProcessingStatus = "OnChain"
 	Invalid      ProcessingStatus = "Invalid"
-	// evaluate adding specific on chain statuses
 )
 
+// Asset structure for TokenFrom, CurrencyTo, and ReStakeTo
+type Asset struct {
```

Comment:
how do you differ "From" types from "To" types
some can be in both but I guess most will be available only in one side